src/frontend/src/eth/components/send/SendAmount.svelte

```diff
@@ -1,91 +1,95 @@
 <script lang="ts">
-	import { Input } from '@dfinity/gix-components';
 	import { FEE_CONTEXT_KEY, type FeeContext } from '$eth/stores/fee.store';
-	import { getContext } from 'svelte';
-	import { debounce, isNullish, nonNullish } from '@dfinity/utils';
-	import { maxGasFee, minGasFee } from '$eth/utils/fee.utils';
-	import { invalidAmount } from '$lib/utils/input.utils';
-	import { parseToken } from '$lib/utils/parse.utils';
+	import { getContext, onDestroy } from 'svelte';
+	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { getFee, maxGasFee, minGasFee } from '$eth/utils/fee.utils';
 	import { BigNumber } from '@ethersproject/bignumber';
-	import { slide } from 'svelte/transition';
 	import { SEND_CONTEXT_KEY, type SendContext } from '$icp-eth/stores/send.store';
 	import { balancesStore } from '$lib/stores/balances.store';
 	import { isSupportedEthTokenId } from '$eth/utils/eth.utils';
 	import { i18n } from '$lib/stores/i18n.store';
 	import type { Token } from '$lib/types/token';
+	import SendInputAmount from '$lib/components/send/SendInputAmount.svelte';
+	import { InsufficientFundsError } from '$lib/types/send';
+	import { getMaxTransactionAmount } from '$lib/utils/token.utils';
 
 	export let amount: number | undefined = undefined;
 	export let insufficientFunds: boolean;
 	export let nativeEthereumToken: Token;
 
-	let insufficientFundsError: string | undefined;
+	let insufficientFundsError: InsufficientFundsError | undefined = undefined;
 
 	$: insufficientFunds = nonNullish(insufficientFundsError);
 
 	const { feeStore: storeFeeData } = getContext<FeeContext>(FEE_CONTEXT_KEY);
 	const { sendTokenDecimals, sendBalance, sendTokenId } = getContext<SendContext>(SEND_CONTEXT_KEY);
 
-	const validate = () => {
-		if (invalidAmount(amount)) {
-			insufficientFundsError = undefined;
+	let fee: BigNumber | undefined | null = undefined;
+
+	let timer: NodeJS.Timeout | undefined;
+
+	$: $storeFeeData,
+		(() => {
+			timer = setTimeout(() => {
+				fee = getFee($storeFeeData);
+			}, 500);
+		})();
+
+	onDestroy(() => {
+		if (isNullish(timer)) {
 			return;
 		}
 
+		clearInterval(timer);
+	});
+
+	$: customValidate = (userAmount: BigNumber) => {
 		if (isNullish($storeFeeData)) {
 			insufficientFundsError = undefined;
 			return;
 		}
 
-		const userAmount = parseToken({
-			value: `${amount}`,
-			unitName: $sendTokenDecimals
-		});
-
 		// If ETH, the balance should cover the user entered amount plus the min gas fee
 		if (isSupportedEthTokenId($sendTokenId)) {
 			const total = userAmount.add(minGasFee($storeFeeData));
 
 			if (total.gt($sendBalance ?? BigNumber.from(0n))) {
-				insufficientFundsError = $i18n.send.assertion.insufficient_funds_for_gas;
-				return;
+				insufficientFundsError = new InsufficientFundsError(
+					$i18n.send.assertion.insufficient_funds_for_gas
+				);
+				return insufficientFundsError;
 			}
 
 			insufficientFundsError = undefined;
-			return;
+			return insufficientFundsError;
```

Comment:
Can you try following:

1. In `SendAmount` export `error`

```
export let error: Error | undefined;
```

2. Bind the error in `SendAmount`

```
<SendInputAmount bind:error={insufficientFundsError} />
```

3. Clean up this function `customValidate` by not assigning a local state and returning error but, only the latest

```
insufficientFundsError = undefined; // <--- remove those
return insufficientFundsError;
```

This can avoid having to assign the value localy and returning it as well.

Reply:
Done